src/custom/components/AccountDetails/Transaction.tsx

```diff
@@ -215,59 +511,96 @@ export default function Transaction({ hash: id }: { hash: string }) {
+                    <SVG src={TxArrowsImage} description="Order Open" />
+                  )}
+                </IconWrapper>
+              </IconType>
+            )}
+            <TransactionStatusText>
+              <ActivitySummary activityData={activityData} id={id} isCancelled={isCancelled} />
+            </TransactionStatusText>
+          </RowFixed>
+        </TransactionState>
+
+        <StatusLabelWrapper>
+          <StatusLabel color={determinePillColour(status, type)} isPending={isPending || isCancelling}>
+            {isConfirmed ? (
+              <SVG src={OrderCheckImage} description="Order Filled" />
+            ) : isExpired ? (
+              <SVG src={OrderExpiredImage} description="Order Expired" />
+            ) : isCancelled ? (
+              <SVG src={OrderCancelledImage} description="Order Cancelled" />
             ) : (
-              getActivitySummary({ activityData, id })
+              <SVG src={OrderOpenImage} description="Order Open" />
             )}
-          </TransactionStatusText>
-        </RowFixed>
-        <IconWrapper pending={isPending || isCancelling} success={isConfirmed || isCancelled}>
-          {isPending || isCancelling ? (
-            <Loader />
-          ) : isConfirmed ? (
-            <CheckCircle size="16" />
-          ) : isExpired ? (
-            <AlertCircle size="16" />
-          ) : isCancelled ? (
-            <XCircle size="16" />
-          ) : (
-            <Triangle size="16" />
+            {isPending
+              ? 'Open'
+              : isConfirmed && isTransaction
+              ? 'Approved'
+              : isConfirmed
+              ? 'Filled'
+              : isExpired
+              ? 'Expired'
+              : isCancelled
+              ? 'Cancelled'
+              : 'Open'}
+          </StatusLabel>
+
+          {isCancelling && (
+            <StatusLabelBelow isCancelling={isCancelling}>
+              Cancellation <br /> requested...
+            </StatusLabelBelow>
           )}
-        </IconWrapper>
-      </TransactionState>
-      {isCancellable && (
-        <>
-          <LinkStyledButton onClick={onCancelClick}>(cancel)</LinkStyledButton>
-          {showCancelModal && (
-            <CancellationModal
-              orderId={id}
-              summary={activityData.summary}
-              isOpen={showCancelModal}
-              onDismiss={onDismiss}
-            />
+
+          {isCancellable && (
+            <StatusLabelBelow>
+              <LinkStyledButton onClick={onCancelClick}>Cancel order</LinkStyledButton>
+              {showCancelModal && (
+                <CancellationModal
+                  orderId={id}
+                  summary={activityData.summary}
+                  isOpen={showCancelModal}
+                  onDismiss={onDismiss}
+                />
+              )}
+            </StatusLabelBelow>
           )}
-        </>
+        </StatusLabelWrapper>
+      </TransactionWrapper>
+
+      {isUnfillable && (
```

Comment:
@alfetopito Moved your `isUnfillable` to here.

Reply:
Do as you see best. I added only as a reference.